src/syntax/herbie-to-egg-pattern.rkt

```diff
@@ -0,0 +1,86 @@
+#lang racket
+(require "./rules.rkt")
+(provide herbie-pattern->rust-pattern)
+
+
+(define (herbie-pattern->rust-pattern datum)
+  (cond
+    [(list? datum)
+     (string-append
+      "("
+      (symbol->string (first datum))
+     (foldr
+      (lambda (sub-expr acc)
+        (string-append " "
+                       (herbie-pattern->rust-pattern sub-expr)
+                       acc))
+      ""
+      (rest datum))
+     ")")]
```

Comment:
There's a cleaner way to write this:

```
(format "(~a ~a)" (first datum) (string-join (map herbie-pattern->rust-pattern (rest datum)) " "))
```

You can also avoid the ugly `(first datum)` by changing the `cond` to a `match`, so you can name the parts of a list. By the way, I think the highlighted code has a (benign) bug: it produces output like `(f  (b)(c))`, misplacing the space...